src/graphql/data/TopTokens.ts

```diff
@@ -132,18 +132,27 @@ function toContractInput(token: PrefetchedTopToken) {
   }
 }
 
-// Map of key: ${chain} + ${address} and value: TopToken object.
+// Map of key: ${HistoryDuration} and value: another Map, of key:${chain} + ${address} and value: TopToken object.
 // Acts as a local cache.
-const tokensWithPriceHistoryCache: Record<string, TopToken> = {}
 
-const checkIfAllTokensCached = (tokens: PrefetchedTopToken[]) => {
+const tokensWithPriceHistoryCache: Record<HistoryDuration, Record<string, TopToken>> = {
+  DAY: {},
+  HOUR: {},
+  MAX: {},
+  MONTH: {},
+  WEEK: {},
+  YEAR: {},
+  '%future added value': {},
+}
+
+const checkIfAllTokensCached = (duration: HistoryDuration, tokens: PrefetchedTopToken[]) => {
   let everyTokenInCache = true
   const cachedTokens: TopToken[] = []
 
   const checkCache = (token: PrefetchedTopToken) => {
     const tokenCacheKey = !!token ? `${token.chain}${token.address}` : ''
-    if (tokenCacheKey in tokensWithPriceHistoryCache) {
-      cachedTokens.push(tokensWithPriceHistoryCache[tokenCacheKey])
+    if (duration in tokensWithPriceHistoryCache && tokenCacheKey in tokensWithPriceHistoryCache[duration]) {
```

Comment:
will `duration in tokensWithPriceHistoryCache` always return true now that you've got a default value?

Reply:
yes true, fixed here and line 215. ty for catching.